benchmarks/linalg_refactor_benchmark.cpp

```diff
@@ -0,0 +1,151 @@
+#include <shogun/base/init.h>
+#include <shogun/lib/SGVector.h>
+#include <shogun/mathematics/linalgrefactor/linalgRefactor.h>
+
+#include <shogun/mathematics/eigen3.h>
+#include <viennacl/linalg/inner_prod.hpp>
+#include <viennacl/vector.hpp>
+
+#include <algorithm>
+#include <memory>
+#include <hayai/hayai.hpp>
+
+#include <iostream>
+
+using namespace shogun;
+
+/**
+ * Instructions :
+ * 1. Install benchmarking toolkit "hayai" (https://github.com/nickbruun/hayai)
+ * 2. Compile against libhayai_main, e.g.
+ * 		g++ -O3 -std=c++11 linglg_refactor_benchmark.cpp -I/usr/include/eigen3 \
+ *		-lshogun -lhayai_main -lOpenCL -o benchmark
+ * 3. ./benchmark
+ */
+
+/** Generate data only once */
+typedef viennacl::vector_base<int32_t, std::size_t, std::ptrdiff_t> VCLVectorBase;
+
+struct Data
```

Comment:
If you create this as a template, then we can easily check with the tests with different data-types. For example, the real use cases would mostly be `float64_t` and `float32_t`. can you please do that?